src/solver/optimisation/constraints/PMaxDispatchableGeneration.cpp

```diff
@@ -1,38 +1,32 @@
 #include "PMaxDispatchableGeneration.h"
 
-void PMaxDispatchableGeneration::add(int pays,
-                                     int cluster,
-                                     int clusterIndex,
-                                     int pdt,
-                                     bool Simulation)
+void PMaxDispatchableGeneration::add(int pays, std::shared_ptr<StartUpCostsData> data)
```

Comment:
```suggestion
void PMaxDispatchableGeneration::add(int pays, StartUpCostsData& data)
```

Reply:
can't commit the suggestion as it, the call and data construction must be changed. I'll take care of that right now